index.js

```diff
@@ -25,27 +25,53 @@ function compose (middleware) {
   }
 
   /**
+   * Make an iterator for middleware.
+   */
+
+  middleware[Symbol.iterator] = function () {
+    const self = this
+    const length = this.length
+    let i = 0
+    let context
+    let nextFunc
+
+    return {
+      next (c, n) {
+        if (!context) context = c
+        if (!nextFunc) nextFunc = n
+
+        let fn = self[i++] || nextFunc
+        let done = i > length
```

Comment:
At `length`, there is no more values and therefore the iterator is done. That means it should be `i >= length`, not `i > length`. However, we also sometimes have `nextFunc`. Therefore it should be `i >= length + !!nextFunc`, essentially adding one to the length for that calculation. Or maybe instead it should be `let length = this.length + !!nextFunc`, this is the only use of the length variable.